voto/bin/add_sailbuoy.py

```diff
@@ -114,17 +114,29 @@ def add_nrt_sailbuoy(df_in, sb, mission):
     _log.info(f"plotting sailbuoy data from SB{sb} mission {mission}")
     sailbuoy_nrt_plots(ds)
     make_map(ds)
-    if df_in.Leak.any() or df_in.BigLeak.any():
-        leak_alert_email(ds)
+    send_alert_email(ds)
     _log.info(f"Completed add SB{sb} mission {mission}")
 
-
-def leak_alert_email(ds_mission):
-    sb_num = ds_mission.attrs["sailbuoy_serial"]
-    msg = f"Leak detected in Sailbuoy {sb_num}"
-    mailer(msg, leak_mails)
-
-
+def send_alert_email(df_in):
+    sb_num = df_in.attrs["sailbuoy_serial"]
+    msg=[]
+    subset_data = df_in[-15:]
+    if df_in.Leak[-15:].any() or df_in.BigLeak.Leak[-15:].any():
+        msg_l = f"Leak detected in Sailbuoy {sb_num}"
+    if len(df_in.Leak[-15:].unique())==1 or len(df_in.BigLeak.Leak[-15:].unique()) == 1:
+        msg_l = str()
+    if df_in.Warning[-15:].any():
+        msg_w = f"There is a warning for Sailbuoy {sb_num}"
+    if len(df_in.Warning[-15:].unique())==1:
+        msg_w = str()
+    if not df_in.WithinTrackRadius[-15:].any():
+        msg_t = f"The Sailbuoy {sb_num} is off track"
+    if len(df_in.WithinTrackRadius[-15:].unique())==1:
+        msg_w = str()
+    msg = "\n".join([msg_l, msg_w,msg_t])
+    if len(msg) > 2:
```

Comment:
with the \n character, won't the message always be at least 6 characters long? So it will always mail?

Reply:
I tested that and it gives len 2. I see your point though